torch/csrc/autograd/python_function.cpp

```diff
@@ -111,7 +124,8 @@ auto PyFunction::legacy_apply(const variable_list& inputs) -> variable_list {
   });
 }
 
-// NOTE: this function is written in a way that assumes it's only called for backward
+// NOTE: this function is written in a way that assumes it's only called for backward;
+// it's used by engine.cpp (NB: this isn't the apply method on autograd Function)
```

Comment:
autograd Function is very inprecise here -- it IS the apply method of an autograd Function 🙂 it's not the `.apply` visible from Python